nuitka/freezer/ImportDetection.py

```diff
@@ -206,38 +206,16 @@ def _detectImports(command):
 
 
 def _detectEarlyImports():
-    encoding_names = [
-        m[1] for m in pkgutil.iter_modules(sys.modules["encodings"].__path__)
-    ]
-
-    if os.name != "nt":
-        # On posix systems, and posix Python variants on Windows, these won't
-        # work and fail to import.
-        for encoding_name in ("mbcs", "cp65001", "oem"):
-            if encoding_name in encoding_names:
-                encoding_names.remove(encoding_name)
-
-    # Not for startup.
-    for non_locale_encoding in (
-        "bz2_codec",
-        "idna",
-        "base64_codec",
-        "hex_codec",
-        "rot_13",
-    ):
-        if non_locale_encoding in encoding_names:
-            encoding_names.remove(non_locale_encoding)
-
-    import_code = ";".join(
-        "import encodings.%s" % encoding_name
-        for encoding_name in sorted(encoding_names)
+    encodings_subs = ";".join(
+        "import encodings.%s" % m[1]
+        for m in pkgutil.iter_modules(sys.modules["encodings"].__path__)
     )
 
-    import_code += ";import locale;"
+    import_code = encodings_subs + ";import locale"
```

Comment:
why we add `locale`?

Reply:
The `locale` is used to react to the user settings for locale, which may not be used for the config Nuitka is running it. And the `inspect` needs to be monkey patched before it's loaded, to become tolerant of compiled  code. And `importlib` is needed to register Nuitka's loader with the application, without it, cannot register the loader for the package itself, i.e. we cannot load `importlib` if its not already there.

Reply:
> And the `inspect` needs to be monkey patched before it's loaded, to become tolerant of compiled code. And `importlib` is needed to register Nuitka's loader with the application, without it, cannot register the loader for the package itself, i.e. we cannot load `importlib` if its not already there.

Correct, I noticed later when I tried to hard-code ignore `inspect`, I got `ModuleNotFoundError` at runtime.



> The `locale` is used to react to the user settings for locale, which may not be used for the config Nuitka is running it.

I couldn't understand, is it only for specified python versions?(I tried to ignore it in py37 and everything are well)